tests/ui/lint/improper-ctypes/mustpass-non-recursion-limit.rs

```diff
@@ -0,0 +1,32 @@
+//@ check-pass
+
+#![recursion_limit = "5"]
+#![allow(unused)]
+#![deny(improper_ctypes)]
+
+#[repr(C)]
+struct F1(*const ());
+#[repr(C)]
+struct F2(*const ());
+#[repr(C)]
+struct F3(*const ());
+#[repr(C)]
+struct F4(*const ());
+#[repr(C)]
+struct F5(*const ());
+#[repr(C)]
+struct F6(*const ());
+
+#[repr(C)]
+struct B {
+    f1: F1,
+    f2: F2,
+    f3: F3,
+    f4: F4,
+    f5: F5,
+    f6: F6,
+}
+
+extern "C" fn foo(_: B) {}
+
+fn main() {}
```

Comment:
What exactly is this expected to test? There doesn't seem to be any recursion here.

Reply:
Maybe this would be good to run twice with different limits:

```rust
//@ revisions: limit5 limit10
#![deny(improper_ctypes)]
#![cfg_attr(limit6, recursion_limit = "10")]
#![cfg_attr(limit5, recursion_limit = "5")]
```

Then add something that passes with the limit of 10 but needs `#[cfg_attr(limit5, expect(improper_ctypes))]` to pass with a limit of 5.

Reply:
ah got it.
it's linked to pull #130758, so the test is here to check that the next attempt to add a recursion limit wouldn't make the same mistakes as the previous one